proto/data.go

```diff
@@ -372,12 +368,11 @@ func (v *Value) GetInt() (int64, error) {
 // GetTime decodes a time value from the bytes field of the receiver. If the
 // tag is not TIME an error will be returned.
 func (v *Value) GetTime() (time.Time, error) {
-	var t time.Time
-	if v.GetTag() != ValueType_TIME {
-		return t, fmt.Errorf("value type is not TIME: %s", v.GetTag())
+	if tag := v.GetTag(); tag != ValueType_TIME {
+		return time.Time{}, fmt.Errorf("value type is not TIME: %s", tag)
 	}
-	err := t.UnmarshalBinary(v.Bytes)
-	return t, err
+	_, t := encoding.DecodeTime(v.Bytes)
```

Comment:
It's important to remember that `proto.Value` carries a checksum, but we don't seem to be using it consistently at all.